samples/bluetooth/df_per_adv_beacon/CMakeLists.txt

```diff
@@ -0,0 +1,39 @@
+#
+# Copyright (c) 2021 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+
+cmake_minimum_required(VERSION 3.13.1)
+
+if (NOT DF_MODE)
+  message("DF_MODE is not being defined on the CMake command-line in the environment or by the app\n"
+          "Setting DF_MODE to default value: AOD")
+  set(DF_MODE "AOD")
+endif()
```

Comment:
It seems to me that it can't be done that way because there is a de-selected entry: https://github.com/zephyrproject-rtos/zephyr/blob/3dd35c202be239884a913d78780eda45ca3c810f/samples/bluetooth/df_per_adv_beacon/df_aoa.conf#L6

Although I can add Kconfig file with two entries `DF_AOA_MODE`, `DF_AOD_MODE` and according to selected one include appropriate *.conf file. 
I'd like to leave separate conf files because it provides complete Kconfig required to use:
- AoA mode
- AoA and AoD modes

Reply:
Yeah, everything is `default y`, maybe they shouldn't have been?

You can also change the default locally in the Kconfig, and add default n.

I think the best would have been that the features didn't default to 'y' and you only turned them on when you actually needed to, if the prj.conf have to turn off all these features it's not a good default selection IMO.

Doing prj.conf manipulation is not a good idea as far as I know, but maybe @tejlmand has a different opinion on this?

Reply:
Supported DF features are set by default to y. Use of e.g. only AoA mode is restricting functionality e.g. to decrease code size. So if one enables DF all supported features are available. From sample application point of view this is most desired configuration. I've provided AoA just to show that there is an option to do so.

Reply:
@tejlmand have you got a suggestion about it?

Reply:
I see you write following in the docs:
> There are two build configuration provided with this sample:
> - df_aoa.config - enables Angle of Arrival mode
> - df_aod.config - enables Angle of Departure mode (this configuration requires antenna matrix)
>
> By default during build process df_aod.conf configuration is used.

This means that the settings in `df_aod.conf` should go directly into `prj.conf` and the config changes needed for `AoA` mode should go into a Kconfig fragment, for example `overlay-aoa.conf`.

That means you can remove:
```
if (NOT DF_MODE)
  message("DF_MODE is not being defined on the CMake command-line in the environment or by the app\n"
          "Setting DF_MODE to default value: AOD")
  set(DF_MODE "AOD")
endif()

if (DF_MODE STREQUAL "AOA")
  set(DF_CONF_FILE "${CMAKE_CURRENT_SOURCE_DIR}/df_aoa.conf")
elseif (DF_MODE STREQUAL "AOD")
  set(DF_CONF_FILE "${CMAKE_CURRENT_SOURCE_DIR}/df_aod.conf")
else()
  message(FATAL_ERROR "Selected unsupported DF_MODE: ${DF_MODE}.\n"
          "Please set DF_MODE parameter to one of following: AOA, AOD.")
endif()

if(NOT EXISTS ${DF_CONF_FILE})
  message(FATAL_ERROR "No such file or directory: DF_CONF_FILE: '${DF_CONF_FILE}'")
endif()
list(APPEND CONF_FILE ${DF_CONF_FILE})
list(APPEND CONF_FILE "prj.conf")
```

and then in the documentation simply write:
> To use Angle of Arrival mode, build this application as follows:
> `west build -b nrf52833dk_nrf52833  samples/bluetooth/df_per_adv_beacon -- -DOVERLAY_CONFIG=overlay-aoa.conf`

For more inspiration, you may take a look at echo server:
https://github.com/zephyrproject-rtos/zephyr/tree/master/samples/net/sockets/echo_server
https://docs.zephyrproject.org/latest/samples/net/sockets/echo_server/README.html

Reply:
@tejlmand thank you! I didn't know about DOVERLAY_CONFIG. It looks like I was trying reinvent the wheel. I'll change it as suggested.